hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/internal/MultiTenantConnectionProviderInitiator.java

```diff
@@ -41,7 +45,36 @@ public Class<MultiTenantConnectionProvider<?>> getServiceInitiated() {
 	@Override
 	public MultiTenantConnectionProvider<?> initiateService(Map<String, Object> configurationValues, ServiceRegistryImplementor registry) {
 		if ( !configurationValues.containsKey( AvailableSettings.MULTI_TENANT_CONNECTION_PROVIDER ) ) {
-			// nothing to do, but given the separate hierarchies have to handle this here.
+			final BeanContainer beanContainer = Helper.allowExtensionsInCdi( registry ) ? registry.requireService( ManagedBeanRegistry.class ).getBeanContainer() : null;
```

Comment:
I think a similar change should be done for `ConnectionProviderInitiator`

Reply:
In practice most of applications are using `DatasourceConnectionProviderImpl`, the underlying `DataSource` is managed bean, I don't think we should add complexity here.

Reply:
I'm inclined to keep it as is for `ConnectionProviderInitiator`, WDYT @gavinking

Reply:
I guess naively I would have said it makes sense to treat `ConnectionProvider`s and `MultiTenantConnectionProvider`s uniformly, but I'm not sure.

Reply:
I appended a commit, please review. @gavinking @beikov